build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/BuildMonitorView.java

```diff
@@ -143,11 +161,28 @@ private boolean isGiven(String value) {
 
     private List<JobView> jobViews() {
         JobViews views = new JobViews(new StaticJenkinsAPIs(), currentConfig());
+        List<JobView> jobs = new ArrayList<JobView>();
 
         //A little bit of evil to make the type system happy.
         @SuppressWarnings("unchecked")
+        List<ItemGroup<?>> groups = new ArrayList(filter(super.getItems(), AbstractFolder.class));
+
+        Pattern includePattern = Strings.isNullOrEmpty(currentConfig().getInclude()) ? null : Pattern.compile(currentConfig().getInclude());
+        Pattern excludePattern = Strings.isNullOrEmpty(currentConfig().getExclude()) ? null : Pattern.compile(currentConfig().getExclude());
```

Comment:
Could we avoid all the null checks here and delegate the decision of whether or not a job should be included in the view to some other class?